internal/k8s/k8s.go

```diff
@@ -244,11 +262,52 @@ func New(cfg *Config) (*Client, error) {
 	}
 
 	if cfg.EnableWebhook {
-		err := enableWebhook(mgr, cfg.ValidateConfig, cfg.Namespace, cfg.Logger)
-		if err != nil {
-			level.Error(c.logger).Log("error", err, "unable to create", "webhooks")
-			return nil, err
+		setupFinished := make(chan struct{})
+		if cfg.EnableCertRotation {
+			webhooks := []rotator.WebhookInfo{
+				{
+					Name: validatingWebhookName,
+					Type: rotator.Validating,
+				},
+				{
+					Name: addresspoolConvertingWebhookCRD,
+					Type: rotator.CRDConversion,
+				},
+				{
+					Name: bgppeerConvertingWebhookCRD,
+					Type: rotator.CRDConversion,
+				},
+			}
+
+			level.Info(c.logger).Log("op", "startup", "action", "setting up cert rotation")
+			err := rotator.AddRotator(mgr, &rotator.CertRotator{
+				SecretKey: types.NamespacedName{
+					Namespace: cfg.Namespace,
+					Name:      webhookSecretName,
+				},
+				CertDir:        cfg.CertDir,
+				CAName:         caName,
+				CAOrganization: caOrganization,
+				DNSName:        fmt.Sprintf("%s.%s.svc", cfg.CertServiceName, cfg.Namespace),
+				IsReady:        setupFinished,
+				Webhooks:       webhooks,
+			})
+			if err != nil {
+				level.Error(c.logger).Log("error", err, "unable to set up", "cert rotation")
+				return nil, err
+			}
+		} else {
+			close(setupFinished)
 		}
+
+		go func() {
+			// Block until the setup (certificate generation) finishes.
+			<-setupFinished
+			err := enableWebhook(c.mgr, cfg.ValidateConfig, cfg.Namespace, cfg.Logger)
```

Comment:
I'd log here that we are enabling the webhooks, so we can see when it happens

Reply:
there is such log in the enableWebhook function